{{cookiecutter.repository_name}}/.github/workflows/tests.yml

```diff
@@ -0,0 +1,65 @@
+name: Run Tests
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: {{ '${{ matrix.os }}' }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.5, 3.6, 3.7, 3.8]
+        os: [ubuntu-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python {{ '${{ matrix.python-version }}' }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: {{ '${{ matrix.python-version }}' }}
+        
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+
+    - name: Test with tox
+      run: tox -e py
+
+{%- if cookiecutter.use_codecov_with_ci == 'y' %}
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        file: ./coverage.xml
+{%- endif %}
+
+{%- if cookiecutter.use_pypi_with_ci == 'y' %}
+  # Automatically deploy releases to PyPI for each tagged commit. This
+  # uses the API Token feature (https://pypi.org/help/#apitoken). After
+  # obtaining an API token from PyPI, add it to your github secrets with
+  # the name `pypi_password`.
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine
```

Comment:
even though its suboptimal to need to install the package, I think it is more robust to do `pip install .[dev]` - for example, if requires to build or publish the distribution change, or of minimum versions need to be increased

Reply:
good point